### PR TITLE
feat(server): Add PUT /:brotherID endpoint on backend

### LINKED ISSUE #40: Create PUT /:brotherID endpoint on backend
Create a PUT (edit existing information) endpoint in the `brothers.controller.js` file that
- Can be accessed at `/:brotherID` endpoint
  - REST means we can make a PUT request to this endpoint, meaning we're updating info about a certain bro
  - So on the frontend we visit something like `sjsuthetatau.com/brothers/edit/:brotherID`, which will make an API call to our server at `/api/brothers/:brotherID`, which is what this task covers
- Uses mongoose to query using the ID url parameter `Brother.findById( req.params.brotherID )`
- Updates the fields with whatever is in the request body (updated fields TBD)
- Saves the changes from the user (which writes to the DB)

Example (check out `router.put('/:auctionID')`):
https://github.com/williamtnguyen/auction-app/blob/master/routes/api/auctions.js

### PR DESCRIPTION
**Pull request by donieypon**
Added a PUT endpoint on backend, still need to discuss on what fields
brothers can edit (e.g., password, bio, etc.).

Closes #40 :)

### CODE REVIEW

**Comment by donieypon** [comment]
Can you guys check what I got so far and lmk what you should be updated? Thanks.

**Review by dtuckersuh — APPROVED** [review]

**Comment by dtuckersuh** [comment]
Can't think of anything else that should be updatable. Maybe email, major, and/or position?

**Comment by williamtnguyen** [comment]
> Can't think of anything else that should be updatable. Maybe email, major, and/or position?

I think those make sense to add, graduateYear as well!

**Comment by donieypon** [comment]
Yeah those seem like good fields to add, should those be implemented now or after we develop the individual brother page?

**Comment by williamtnguyen** [comment]
> Yeah those seem like good fields to add, should those be implemented now or after we develop the individual brother page?

Now, go ahead and make those fields the ones that are updated. Force push your changes when done

File: routes/api/brothers/brothers.controller.js [hunk 1]
```diff
@@ -111,4 +111,28 @@ brotherController.post('/login', (req, res) => {
...
 });
 
+/**
+ * EDIT Endpoint
+ * @route PUT api/borthers/edit
+ * @desc edit a bro page
+ */
+brotherController.put('/:brotherID', (req, res) => {
+  // Form validation
+  const { errors, isValid } = validateLoginInput(req.body);
```
Comment (williamtnguyen) [inline routes/api/brothers/brothers.controller.js:121]: This function can't be used to validate edit form, it's for login. You'll need to write a validation function for the edit page individually, let me know if you need help w this. It's a bit tricky bc users won't edit every field most of the time, so we have to account for empty inputs accordingly

File: routes/api/brothers/brothers.controller.js [hunk 2]
```diff
@@ -111,4 +111,28 @@ brotherController.post('/login', (req, res) => {
...
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+
+  Brother.findById(req.params.brotherID, (err, foundBrother) => {
+    if (err) {
+      return console.log('Could not find the Brother in the DB: ${err}')
+    }
+
+    // Form fields:
```
Comment (williamtnguyen) [inline routes/api/brothers/brothers.controller.js:131]: Potentially update password, biography, email, major, position, and graduateYear depending which ones are not empty from the frontend

**Review by williamtnguyen — CHANGES_REQUESTED** [review]

**Review by donieypon — COMMENTED** [review]
_yoooo <3_

File: routes/api/util/form-validation/edit.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const Validator = require('validator');
+const isEmpty = require('is-empty');
+
+/**
+ * Takes in edit form input and checks for validity
+ * @param requestBody the data object holding all form inputs as fields
+ */
+function validateEditInput(requestBody) {
```
Comment (williamtnguyen) [inline routes/api/util/form-validation/edit.js:8]: One thing I just realized is if they submit the form and all inputs are empty, then nothing would update and we'd write to MongoDB for no reason. Maybe loop through the `requestBody` before checking any of the fields and iterate an `emtpyCount` variable, and if the emptyCount is the same as the requestBody size then we can add an error that says everything empty

File: routes/api/util/form-validation/edit.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const errors = {};
+  // Only validate email and gradYear because they have more strict input
+  if (!Validator.isEmail(requestBody.email)) {
```
Comment (williamtnguyen) [inline routes/api/util/form-validation/edit.js:11]: Also for all of these we'd have to check if they're not empty before considering validating them like
```
if (!isEmpty(requestBody.email) && !Validator.isEmail(requestBody.email))
```

**Review by williamtnguyen — CHANGES_REQUESTED** [review]
Left some comments!

**Comment by dtuckersuh** [comment]
Fixed the edit validation details. Lmk what yall think!

**Review by williamtnguyen — APPROVED** [review]
🔥

**Pull request merged**